src/libponyc/expr/call.c

```diff
@@ -240,6 +240,10 @@ static bool check_arg_types(pass_opt_t* opt, ast_t* params, ast_t* positional,
     {
       errorframe_t frame = NULL;
       ast_error_frame(&frame, arg, "argument not a subtype of parameter");
+      ast_error_frame(&frame, arg, "argument type is %s",
+                      ast_print_type(a_type));
+      ast_error_frame(&frame, arg, "parameter type is %s",
```

Comment:
It would be better to pass `param` as the second argument here, so that the parameter declaration is printed alongside the type. Same thing for the changes in `ffi.c` and `lambda.c`.